packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/version.dart';
+
+// Include VS Code insiders (useful for debugging).
+const bool _includeInsiders = false;
+
+class VsCode {
+  static const String extensionIdentifier = 'Dart-Code.dart-code';
+
+  VsCode._(this.directory, this.extensionDirectory, { Version version })
+      : this.version = version ?? Version.unknown {
+    _init();
+  }
+
+  final String directory;
+  final String extensionDirectory;
+  final Version version;
+
+  bool _isValid = false;
+  Version extensionVersion;
```

Comment:
Should this be publicly writable?

Reply:
Changed in 4f8ebf4 - it should've been completely private (I'm used to C# with private being default so this still catches me out a little!).